tests/Integration/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCaseTest.php

```diff
@@ -288,6 +277,67 @@ public function testGivenTransactionIdInBookedChildDonation_noNewDonationIsCreat
 		$this->assertFalse( $useCase->handleNotification( $request ) );
 	}
 
+	public function testWhenNotificationIsForNonExistingDonation_newDonationIsCreated() {
+		$repositorySpy = new DonationRepositorySpy();
+
+		$request = ValidPayPalNotificationRequest::newInstantPaymentForDonation( 12345 );
+		$useCase = new HandlePayPalPaymentNotificationUseCase(
+				$repositorySpy,
+			new SucceedingDonationAuthorizer(),
+			$this->getMailer(),
+			new NullLogger()
+		);
+
+		$useCase->handleNotification( $request );
+
+		$storeDonationCalls = $repositorySpy->getStoreDonationCalls();
+		$this->assertCount( 1, $storeDonationCalls, 'Donation is stored' );
+
+		/** @var Donation $donation */
+		$donation = $storeDonationCalls[0];
+		$this->assertEquals( 0, $donation->getPaymentIntervalInMonths(), 'Payment interval is always empty' );
+
+		$this->assertNull( $donation->getId(), 'ID is not taken from request' );
+
+		$donorName = $donation->getDonor()->getPersonName();
+		$this->assertEquals( PersonName::PERSON_PRIVATE, $donorName->getPersonType(), 'Person is always private' );
+		$this->assertEquals( ValidPayPalNotificationRequest::PAYER_ADDRESS_NAME, $donorName->getFullName() );
+
+		$this->assertEquals( ValidPayPalNotificationRequest::PAYER_EMAIL, $donation->getDonor()->getEmailAddress() );
+
+		$address = $donation->getDonor()->getPhysicalAddress();
+		$this->assertEquals( ValidPayPalNotificationRequest::PAYER_ADDRESS_STREET, $address->getStreetAddress() );
+		$this->assertEquals( ValidPayPalNotificationRequest::PAYER_ADDRESS_CITY, $address->getCity() );
+		$this->assertEquals( ValidPayPalNotificationRequest::PAYER_ADDRESS_POSTAL_CODE, $address->getPostalCode() );
+		$this->assertEquals( ValidPayPalNotificationRequest::PAYER_ADDRESS_COUNTRY_CODE, $address->getCountryCode() );
+
+		$payment = $donation->getPayment();
+		$this->assertEquals( ValidPayPalNotificationRequest::AMOUNT_GROSS_CENTS, $payment->getAmount()->getEuroCents() );
```

Comment:
All of those assertEquals can be put into a more high level assertion method, so the details don't clutter the test method

Also don't forget that when comparing scalar values, `assertSame` is better